### ISSUE TITLE
Faster `pad_sequence` and `tensor_split` function with CUDA kernel, are they possible?

### ISSUE THREAD

**Issue by liuzhy71** [pytorch/pytorch#94167]
### 🚀 The feature, motivation and pitch

I was using `torch.nn.utils.rnn.pad_sequence()` to pad a list of variable length tensors. However, I find the padding operation is not so fast. Although I found that this operation has already been moved to C++ side for shorter run time. It is still slow compared to other operations with CUDA kernel support.

(I briefly profiled the computation to see this. Data size about 4000 1D tensors after spliting, maximum vector length about 200 float64 numbers. Running time without profiler 0.05s.
![image](https://user-images.githubusercontent.com/16524611/216874366-710a703b-e0ae-477b-bb4c-c0b2b80996b5.png)

Is it possible to get a CUDA kernel support version of these operation? I found a possible implementation [here](https://github.com/frostblue-wukong/Paddle-Lite/commit/126691f01f96aa0dbe34417165cb0cb09b7e557a), but I am not handy in CUDA kernel. Could you add a feature for this?

### Alternatives

I have tried pad_sequence and tensor_split with GPU tensors and CPU tensors, and I found that CPU tensors have a shorter runing time. So I guess that C++ side padding and splitting does not have a CUDA kernel yet?

**Comment by ezyang** [comment]
PT2 compilation might be able to help, not sure exactly how though

**Comment by liuzhy71** [comment]
> PT2 compilation might be able to help, not sure exactly how though
I have tried torch.jit.script for the process. I haven't try PT2 yet. Is torch.compile() the same as torch.jit.script. Does it provide parallelism?

**Comment by ezyang** [comment]
It's a different system. torch.compile doesn't currently do parallel execution, though it could
